datasets/ar_res_reviews/ar_res_reviews.py

```diff
@@ -0,0 +1,88 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Dataset of 8364 restaurant reviews scrapped from qaym.com in Arabic for sentiment analysis"""
+
+from __future__ import absolute_import, division, print_function
+
+import csv
+
+import datasets
+
+
+# import json
+# import os
+
+
+_CITATION = """\
+@InProceedings{10.1007/978-3-319-18117-2_2,
+author="ElSahar, Hady
+and El-Beltagy, Samhaa R.",
+editor="Gelbukh, Alexander",
+title="Building Large Arabic Multi-domain Resources for Sentiment Analysis",
+booktitle="Computational Linguistics and Intelligent Text Processing",
+year="2015",
+publisher="Springer International Publishing",
+address="Cham",
+pages="23--34",
+isbn="978-3-319-18117-2"
+}
+"""
+
+_DESCRIPTION = """\
+Dataset of 8364 restaurant reviews scrapped from qaym.com in Arabic for sentiment analysis
+"""
+
+_HOMEPAGE = "https://github.com/hadyelsahar/large-arabic-sentiment-analysis-resouces"
+
+_DOWNLOAD_URL = (
+    "https://raw.githubusercontent.com/hadyelsahar/large-arabic-sentiment-analysis-resouces/master/datasets/RES1.csv"
+)
+
+
+class ArResReviews(datasets.GeneratorBasedBuilder):
+    """Dataset of 8364 restaurant reviews in Arabic for sentiment analysis"""
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "polarity": datasets.Value("string"),
```

Comment:
Could you use the ClassLabel feature type for the polarity field ?
For example
```
"polarity": datasets.ClassLabel(names=["-1", "1"])
```
You can change the names to whatever makes more sense (maybe simply "negative" and "positive")

After this change you'll need to regenerate the dataset_infos.json file

Reply:
Hey @lhoestq I tried to use the ClassLabel feature but it fails the cli test using ```python datasets-cli test datasets/ar_res_reviews --save_infos --all_configs``` and produces KeyError: 'polarity'

Reply:
this is the error I'm getting

```Traceback (most recent call last):
  File "datasets-cli", line 36, in <module>
    service.run()
  File "/home/mesfas/opensource/datasets/src/datasets/commands/test.py", line 113, in run
    builder.download_and_prepare(
  File "/home/mesfas/opensource/datasets/src/datasets/builder.py", line 526, in download_and_prepare
    self._download_and_prepare(
  File "/home/mesfas/opensource/datasets/src/datasets/builder.py", line 604, in _download_and_prepare
    self._prepare_split(split_generator, **prepare_split_kwargs)
  File "/home/mesfas/opensource/datasets/src/datasets/builder.py", line 905, in _prepare_split
    example = self.info.features.encode_example(record)
  File "/home/mesfas/opensource/datasets/src/datasets/features.py", line 799, in encode_example
    return encode_nested_example(self, example)
  File "/home/mesfas/opensource/datasets/src/datasets/features.py", line 709, in encode_nested_example
    return dict(
  File "/home/mesfas/opensource/datasets/src/datasets/features.py", line 710, in <genexpr>
    (k, encode_nested_example(sub_schema, sub_obj)) for k, (sub_schema, sub_obj) in utils.zip_dict(schema, obj)
  File "/home/mesfas/opensource/datasets/src/datasets/features.py", line 737, in encode_nested_example
    return schema.encode_example(obj)
  File "/home/mesfas/opensource/datasets/src/datasets/features.py", line 522, in encode_example
    example_data = self.str2int(example_data)
  File "/home/mesfas/opensource/datasets/src/datasets/features.py", line 481, in str2int
    output.append(self._str2int[str(value)])```

Reply:
This is because the first line of the CSV contains the column names. This row has to be ignored

Reply:
@lhoestq hey, skipped the first line via ```csv_file = csv_file.readlines()[1:]```

I got this error ```Traceback (most recent call last):
  File "./datasets/ar_res_reviews/test.py", line 3, in <module>
    data = load_dataset('./datasets/ar_res_reviews', split = "train")
  File "/home/mesfas/opensource/datasets/src/datasets/load.py", line 607, in load_dataset
    builder_instance.download_and_prepare(
  File "/home/mesfas/opensource/datasets/src/datasets/builder.py", line 526, in download_and_prepare
    self._download_and_prepare(
  File "/home/mesfas/opensource/datasets/src/datasets/builder.py", line 604, in _download_and_prepare
    self._prepare_split(split_generator, **prepare_split_kwargs)
  File "/home/mesfas/opensource/datasets/src/datasets/builder.py", line 905, in _prepare_split
    example = self.info.features.encode_example(record)
  File "/home/mesfas/opensource/datasets/src/datasets/features.py", line 799, in encode_example
    return encode_nested_example(self, example)
  File "/home/mesfas/opensource/datasets/src/datasets/features.py", line 709, in encode_nested_example
    return dict(
  File "/home/mesfas/opensource/datasets/src/datasets/features.py", line 710, in <genexpr>
    (k, encode_nested_example(sub_schema, sub_obj)) for k, (sub_schema, sub_obj) in utils.zip_dict(schema, obj)
  File "/home/mesfas/opensource/datasets/src/datasets/features.py", line 737, in encode_nested_example
    return schema.encode_example(obj)
  File "/home/mesfas/opensource/datasets/src/datasets/features.py", line 522, in encode_example
    example_data = self.str2int(example_data)
  File "/home/mesfas/opensource/datasets/src/datasets/features.py", line 481, in str2int
    output.append(self._str2int[str(value)])
KeyError: '-1'```

I think the main issue is that when I use ClassLabel I ignore defining the feature "polarity" datatype, src/datasets/features.py messes this up but what causes output.append(self._str2int[str(value)]) to fail? if the value is inferred to be int (it is an int in reality), str(data) should take care of that, right?

Reply:
@lhoestq To elaborate, when I define the polarity as a string and use ```"polarity": datasets.Value("string")``` instead of using class label I get no issues at all.

Reply:
hey, I changed the following:

```    def _info(self):
        return datasets.DatasetInfo(
            description=_DESCRIPTION,
            features=datasets.Features(
                {
                    "polarity": datasets.ClassLabel(names=["negative", "positive"]),
                    "polarity": datasets.Value("string"),
                    "text": datasets.Value("string"),
                    "restaurant_id": datasets.Value("string"),
                    "user_id": datasets.Value("string"),
                }
            ),
            homepage=_HOMEPAGE,
            citation=_CITATION,
        )
```

now it works well and I was able to generate dummy data and info.json. But the data for polarity is still -1 or 1, not positive or negative, and flake8 returns the following:

```
(ar_res_reviews) mesfas@x1carbon:~/opensource/datasets$ flake8 datasets
datasets/ar_res_reviews/ar_res_reviews.py:58:21: F601 dictionary key 'polarity' repeated with different values
datasets/ar_res_reviews/ar_res_reviews.py:59:21: F601 dictionary key 'polarity' repeated with different values
```

so either I use class label or define the feature as string value, right?